bot.ml

```diff
@@ -690,107 +699,100 @@ let callback _conn req body =
-        when string_match ~regexp:"@coqbot: [Rr]un CI now" comment_info.body
-        -> (
-        match Map.find owner_team_map comment_info.issue.issue.owner with
-        | Some team when signed ->
-            (fun () ->
-              (let open Lwt_result.Infix in
-              GitHub_queries.get_team_membership ~token:github_access_token
-                ~org:comment_info.issue.issue.owner ~team
-                ~user:comment_info.author
-              >>= (fun is_member ->
-                    if is_member then (
-                      Stdio.printf "Authorized user: pushing to GitLab.\n" ;
-                      match comment_info.pull_request with
-                      | Some pr_info ->
-                          pull_request_updated pr_info ()
-                      | None ->
-                          let {GitHub_subscriptions.owner; repo; number} =
-                            comment_info.issue.issue
-                          in
-                          GitHub_queries.get_pull_request_refs
-                            ~token:github_access_token ~owner ~repo ~number
-                          >>= fun pr_info ->
-                          pull_request_updated
-                            {pr_info with issue= comment_info.issue}
-                            () )
-                    else
-                      Lwt_io.print "Unauthorized user: doing nothing.\n"
-                      |> Lwt_result.ok)
-              |> Fn.flip Lwt_result.bind_lwt_err (fun err ->
-                     Lwt_io.printf "Error: %s\n" err))
-              >>= fun _ -> Lwt.return ())
+          let body = trim_comments body in
+          if
+            string_match
+              ~regexp:(f "@%s:? minimize[^```]*```\\([^```]+\\)```" bot_name)
+              body
+          then
+            run_coq_minimizer ~script:(Str.matched_group 1 body)
+              ~comment_thread_id:issue_info.id ~comment_author:issue_info.user
+            |> Lwt.async ;
+          Server.respond_string ~status:`OK ~body:"Handling minimization." ()
+      | Ok (signed, GitHub_subscriptions.CommentCreated comment_info) ->
+          let body = trim_comments comment_info.body in
+          if
+            string_match
+              ~regexp:(f "@%s:? minimize[^```]*```\\([^```]+\\)```" bot_name)
+              body
+          then (
+            run_coq_minimizer ~script:(Str.matched_group 1 body)
+              ~comment_thread_id:
+                ( match comment_info.pull_request with
+                | None ->
+                    comment_info.issue.id
+                | Some pr ->
+                    pr.issue.id )
+              ~comment_author:comment_info.author
             |> Lwt.async ;
+            Server.respond_string ~status:`OK ~body:"Handling minimization." ()
+            )
+          else if string_match ~regexp:(f "@%s: [Rr]un CI now" bot_name) body
+          && (match comment_info.pull_request with None -> false | Some _ -> true)
```

Comment:
FYI, you have a function `Option.is_some` that does this, cf. 68a18eeb725f9f12ac497d124f022f017a1033cf.